artiq/coredevice/dac34h84.py

```diff
@@ -110,7 +110,7 @@ class DAC34H84:
     syncsel_mixercd = 0b1001  # sif_sync and register write
     syncsel_nco = 0b1000  # sif_sync
     syncsel_fifo_input = 0b10  # external lvds istr
-    sif_sync = 1
+    sif_sync = 0
```

Comment:
Document that the user must not change that.

Reply:
Are you sure? `dac_sync` does set, then clear. If `sif_sync` is set earlier during the mmap writing, then this is set-set-clear and the only rising edge is the first one during mmap-writing. (under the assumptions of https://github.com/m-labs/artiq/pull/1657#discussion_r614159355).

Reply:
Either way, there would be one rising edge. All relevant configuration data is written before sif_sync when writing the dac mmap, either timing is acceptable.

Reply:
But then your entire line of argument about `sif_sync=1` by default being wrong and it not being toggled in init breaks down.
Then let's keep it at 1 by default, remove the `dac_sync()` in init, and make it clear-then-set.
Less code is typically better. And this would also remove the implicit `sif_sync=0` predicate for `dac_sync()`.

Reply:
Ok, I'll update it to clear then set.